api/tests/integration/infrastructure/repositories/participant-result-repository_test.js

```diff
@@ -702,6 +613,45 @@ describe('Integration | Repository | ParticipantResultRepository', function() {
         expect(error).to.be.instanceOf(NotFoundError);
       });
     });
+
+    context('when the participation is shared', function() {
+      it('returns the mastery rate for the participation', async function() {
+        const { id: userId } = databaseBuilder.factory.buildUser();
+        const { id: campaignId } = databaseBuilder.factory.buildCampaign({ targetProfileId });
+        const { id: campaignParticipationId } = databaseBuilder.factory.buildCampaignParticipation({
+          userId,
+          campaignId,
+          isShared: true,
+          sharedAt: new Date('2020-01-02'),
+          masteryPercentage: 0.6,
+        });
+
+        databaseBuilder.factory.buildAssessment({ campaignParticipationId, userId, state: 'completed' });
+
+        await databaseBuilder.commit();
+        const participantResult = await participantResultRepository.getByUserIdAndCampaignId({ userId, campaignId, locale: 'FR' });
+        expect(participantResult.masteryRate).to.equal(0.6);
+      });
+    });
+
+    context('when the participation is not shared', function() {
+      it('returns the mastery rate for the participation', async function() {
```

Comment:
**niptick** : les 2 it ont la même description, peut-être être plus explicite genre en rajoutant `based on knowledge elements` ou `based on masteryPercentage attribute`

Reply:
done